src/prefect/flows.py

```diff
@@ -347,18 +348,23 @@ def __init__(
         # Check for collision in the registry
         registry = PrefectObjectRegistry.get()
 
-        if registry and any(
-            other
-            for other in registry.get_instances(Flow)
-            if other.name == self.name and id(other.fn) != id(self.fn)
+        if (
+            registry
+            and any(
+                other
+                for other in registry.get_instances(Flow)
+                if other.name == self.name and id(other.fn) != id(self.fn)
+            )
+            and PREFECT_WARN_OBJECT_REGISTRY_CONFLICT.value()
```

Comment:
nit: The `.value` isn't needed.
```suggestion
            and PREFECT_WARN_OBJECT_REGISTRY_CONFLICT
```